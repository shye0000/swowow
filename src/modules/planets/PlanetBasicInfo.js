import React from 'react';
import Row from 'antd/lib/row';
import Col from 'antd/lib/col';

const InfoRow = ({label, value}) => {
	return <Row gutter={24} type="flex">
		<Col xs={24} md={12}>{label}</Col>
		<Col xs={24} md={12}>{value}</Col>
	</Row>;
};

const PlanetBasicInfo = ({planet}) => {
	if (!planet) {
		return null;
	}
	const {
		climate,
		diameter,
		gravity,
		orbital_period,
		population,
		rotation_period,
		surface_water,
		terrain
	} = planet;
	const basicInfo = {
		'Rotation period':    rotation_period,
		'Orbital period':     orbital_period,
		'Diameter':           diameter,
		'Surface water':      surface_water,
		'Population':         population,
		'Gravity':            gravity,
		'Climate':            climate,
		'terrain':            terrain
	};
	return <div className="planet-basic-info">
		{Object.keys(basicInfo).map(label => <InfoRow key={label} label={label} value={basicInfo[label]} />)}
	</div>;
};

export default PlanetBasicInfo;