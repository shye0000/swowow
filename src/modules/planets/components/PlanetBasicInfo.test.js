import React from 'react';
import PlanetBasicInfo from './PlanetBasicInfo';
import { shallow } from 'enzyme';

describe('PlanetBasicInfo', () => {
	it('render correctly', () => {
		const props = {
			planet: {
				rotation_period: 100,
				orbital_period: 100,
				diameter: 100,
				surface_water: 100,
				population: 1000000,
				gravity: 'gravity',
				climate: 'climate',
				terrain: 'terrain'
			}
		};
		const wrapper = shallow(<PlanetBasicInfo {...props} />);
		expect(wrapper).toMatchSnapshot();
	});

	it('when planet is null, rendered nothing (render method return false || null || undefined)', () => {
		const props = {};
		const wrapper = shallow(<PlanetBasicInfo {...props} />);
		expect(wrapper.get(0)).toBeFalsy();
	});
});