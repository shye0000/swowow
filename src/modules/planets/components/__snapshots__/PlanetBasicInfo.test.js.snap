// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanetBasicInfo render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlanetBasicInfo
    planet={
      Object {
        "climate": "climate",
        "diameter": 100,
        "gravity": "gravity",
        "orbital_period": 100,
        "population": 1000000,
        "rotation_period": 100,
        "surface_water": 100,
        "terrain": "terrain",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <InfoRow
          label="Rotation period"
          value={100}
        />,
        <InfoRow
          label="Orbital period"
          value={100}
        />,
        <InfoRow
          label="Diameter"
          value={100}
        />,
        <InfoRow
          label="Surface water"
          value={100}
        />,
        <InfoRow
          label="Population"
          value={1000000}
        />,
        <InfoRow
          label="Gravity"
          value="gravity"
        />,
        <InfoRow
          label="Climate"
          value="climate"
        />,
        <InfoRow
          label="terrain"
          value="terrain"
        />,
      ],
      "className": "planet-basic-info",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Rotation period",
        "nodeType": "function",
        "props": Object {
          "label": "Rotation period",
          "value": 100,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Orbital period",
        "nodeType": "function",
        "props": Object {
          "label": "Orbital period",
          "value": 100,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Diameter",
        "nodeType": "function",
        "props": Object {
          "label": "Diameter",
          "value": 100,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Surface water",
        "nodeType": "function",
        "props": Object {
          "label": "Surface water",
          "value": 100,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Population",
        "nodeType": "function",
        "props": Object {
          "label": "Population",
          "value": 1000000,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Gravity",
        "nodeType": "function",
        "props": Object {
          "label": "Gravity",
          "value": "gravity",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Climate",
        "nodeType": "function",
        "props": Object {
          "label": "Climate",
          "value": "climate",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "terrain",
        "nodeType": "function",
        "props": Object {
          "label": "terrain",
          "value": "terrain",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <InfoRow
            label="Rotation period"
            value={100}
          />,
          <InfoRow
            label="Orbital period"
            value={100}
          />,
          <InfoRow
            label="Diameter"
            value={100}
          />,
          <InfoRow
            label="Surface water"
            value={100}
          />,
          <InfoRow
            label="Population"
            value={1000000}
          />,
          <InfoRow
            label="Gravity"
            value="gravity"
          />,
          <InfoRow
            label="Climate"
            value="climate"
          />,
          <InfoRow
            label="terrain"
            value="terrain"
          />,
        ],
        "className": "planet-basic-info",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Rotation period",
          "nodeType": "function",
          "props": Object {
            "label": "Rotation period",
            "value": 100,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Orbital period",
          "nodeType": "function",
          "props": Object {
            "label": "Orbital period",
            "value": 100,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Diameter",
          "nodeType": "function",
          "props": Object {
            "label": "Diameter",
            "value": 100,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Surface water",
          "nodeType": "function",
          "props": Object {
            "label": "Surface water",
            "value": 100,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Population",
          "nodeType": "function",
          "props": Object {
            "label": "Population",
            "value": 1000000,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Gravity",
          "nodeType": "function",
          "props": Object {
            "label": "Gravity",
            "value": "gravity",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Climate",
          "nodeType": "function",
          "props": Object {
            "label": "Climate",
            "value": "climate",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "terrain",
          "nodeType": "function",
          "props": Object {
            "label": "terrain",
            "value": "terrain",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
